### DISCUSSION TITLE
How to protect master branch on solo-developer projects?

### DISCUSSION THREAD

**Discussion by matt-winfield** [community/community#23727] [New to GitHub]
<p>This is a common problem I have in several repos: I would like to protect the master branch on the projects I work on, so that I know every change has to have been built/tested by GitHub actions.</p>
<p>The problem is that since I am the only developer, I cannot get any PRs approved since you can’t approve your own PR. This means I have to enable “Allow administrator override”, which means I could still accidentally push to the master branch.</p>
<p>Is there any way to ensure that I, even as an administrator, cannot push directly to the master branch, and have to have a PR build successfully, but can still merge PRs without having a second developer for approval?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/matt-winfield">@matt-winfield</a>, welcome to the GitHub Support Community!</p>
<p>I think you could set this up by enabling “Require status checks to pass” but <em>not</em> enabling “require pull request reviews”</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/2/d280676977568176c33812e1cc028c2f175f21bf.png" data-download-href="/uploads/short-url/u2bn7N8azkG8RzF7kaAZoqAfGxh.png?dl=1" title="Screenshot 2020-10-15 at 13.54.07"><img src="https://user-images.githubusercontent.com/2439407/181078926-fca63965-6295-4b16-9edf-cf1954a41945.png" alt="Screenshot 2020-10-15 at 13.54.07" data-base62-sha1="u2bn7N8azkG8RzF7kaAZoqAfGxh" width="690" height="185" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/2/d280676977568176c33812e1cc028c2f175f21bf_2_690x185.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/2/d280676977568176c33812e1cc028c2f175f21bf_2_1035x277.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/d/2/d280676977568176c33812e1cc028c2f175f21bf_2_1380x370.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/2/d280676977568176c33812e1cc028c2f175f21bf_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-10-15 at 13.54.07</span><span class="informations">1836×494 66.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>You can then enable “Include administrators” to require these restrictions on yourself</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/3/63c61bb80ff4f6da4418c6e1e62e24b56512763f.png" data-download-href="/uploads/short-url/eeDIiVrPrQdL8KINJxSH85jZ3JB.png?dl=1" title="Screenshot 2020-10-15 at 13.55.17"><img src="https://user-images.githubusercontent.com/2439407/181078928-c1ba6de2-2ccb-40fa-b768-b90be9b7796b.png" alt="Screenshot 2020-10-15 at 13.55.17" data-base62-sha1="eeDIiVrPrQdL8KINJxSH85jZ3JB" width="690" height="58" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/3/63c61bb80ff4f6da4418c6e1e62e24b56512763f_2_690x58.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/3/63c61bb80ff4f6da4418c6e1e62e24b56512763f_2_1035x87.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/3/63c61bb80ff4f6da4418c6e1e62e24b56512763f_2_1380x116.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/3/63c61bb80ff4f6da4418c6e1e62e24b56512763f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-10-15 at 13.55.17</span><span class="informations">1830×154 13.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>That should mean that you’re not blocked from merging pull requests without a review, but you’ll still be blocked if your checks or CI hasn’t finished running.</p>
